src/domain/core/src/services/ingest/data_writer.rs

```diff
@@ -72,10 +77,12 @@ pub struct WriteDataResult {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+/// Do not create directly, only use with [DataWriter::stage].
```

Comment:
This struct lives in domain layer, while being created in infa layer in `ingest-datafusion` crate, so I can't really hide the constructor. Getters can only help to prevent mutation of fields, but that's a lesser concern. In future we can consider some sort of typestate pattern to allow transition to `commit` only after `stage` was called on the writer.